application/lib/service/products_screen_service.dart

```diff
@@ -0,0 +1,26 @@
+import 'package:application/model/product_dto.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+
+class ProductsScreenService {
+  final FirebaseFirestore _firestore = FirebaseFirestore.instance;
+
+  Stream<List<ProductDTO>> fetchProducts(String searchTerm) {
```

Comment:
A fetchProduct nagyon általános kifejezés, itt valójában az történik hogy keresési feltétel alapján le vannak szűrve a product-ok. Ha lenne egy olyan lekérdezésed, ami lekéri az összes productot, akár annak is lehetne a neve fetchProduct. Itt talán job a fetchProductByName